File: Binary Search/Easy/LET_no69_Sqrt(x).cpp
class Solution {
public:
    int mySqrt(int x) {
        long low = 0, high = x;
        while (low + 1 < high) {
            long mid = low + (high - low) / 2;
            if (mid * mid > x)
                high = mid;
            else
                low = mid;
        }

        if (high * high <= x)
            return high;
        return low;
    }
};

File: Binary Search/Easy/LET_no69_Sqrt(x).py
class Solution:
    def mySqrt(self, x: int) -> int:
        left, right = 0, x
        while left + 1 < right:
            mid = (left + right) // 2
            if x <= mid ** 2:
                right = mid
            else:
                left = mid
        
        return right if right ** 2 <= x else left